subprojects/files/src/test/groovy/org/gradle/api/internal/file/collections/DirectoryFileTreeTest.groovy

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.internal.file.collections
+
+import org.gradle.api.file.FileVisitDetails
+import org.gradle.api.file.FileVisitor
+import org.gradle.api.internal.file.TestFiles
+import org.gradle.api.specs.Spec
+import org.gradle.api.tasks.util.PatternSet
+import org.gradle.test.fixtures.AbstractProjectBuilderSpec
+
+class DirectoryFileTreeTest extends AbstractProjectBuilderSpec {
```

Comment:
❌ Can we have this test for `ReproducibleDirectoryWalker` as well?

Reply:
Done

Reply:
<img width="633" alt="Screen Shot 2019-06-25 at 12 08 14" src="https://user-images.githubusercontent.com/1841944/60085769-9716cb80-9742-11e9-8db0-f268963038f3.png">
<img width="790" alt="Screen Shot 2019-06-25 at 12 15 46" src="https://user-images.githubusercontent.com/1841944/60086006-fd9be980-9742-11e9-9d74-9fe65b7d1a36.png">